### DISCUSSION TITLE
with example code getting: Error while retrieving user profil

### DISCUSSION THREAD

**Discussion by leowenlu** [fastapi-users/fastapi-users#1434] [Q&A] [answered]
I am new to this, have tried a few times and had good read with tutorials/docs, but struggling with the example code to make it work. the example code https://github.com/fastapi-users/fastapi-users/tree/master/examples/sqlalchemy-oauth

My frontend React code is like the following:

```
  const onGoogleLogin = async () => {
    // window.location.href = `${Config.API_BASE_URL}${Config.API_GOOGLE_AUTH_URL}`;
    console.log(`login: ${Config.API_BASE_URL}${Config.API_GOOGLE_AUTH_URL}`);
    try {

      const response = await fetch(`http://localhost:8081/auth/google/authorize`);
      const data = await response.json();
      // Redirect to the Google Auth URL
      window.location.href = data.authorization_url;
    } catch (error) {
      console.error('Error initiating Google login:', error);
    }
  };
```
The google test user is able to login and get to redirect to the `authorization_url`, but the backend got the following error:

```
    account_id, account_email = await oauth_client.get_id_email(
  File ".venv/lib/python3.10/site-packages/httpx_oauth/clients/google.py", line 77, in get_id_email
    raise GetIdEmailError(response=response)
httpx_oauth.exceptions.GetIdEmailError: Error while retrieving user profile.
```

**Comment by leowenlu** [comment]
I am reading here: https://github.com/fastapi-users/fastapi-users/blob/master/docs/usage/routes.md#get-callback

> !!! fail "400 Bad Request" The OAuth provider didn't return an e-mail address. Make sure this provider return e-mail address through their API and you have asked for the required scope.

Where I config `scope`? it' the example code, suppose it should setup this already by default?

Tried to print out the following in `get_id_email` function in `httpx_oauth/clients/google.py`

PROFILE_ENDPOINT: https://people.googleapis.com/v1/people/me
response: <Response [403 Forbidden]>

I did check I have all 3 options ticked in 'OAuth consent screen' in Google APIs and services.

**Reply by hgalytoby** [reply]
I tested my project's google login without any problem.

I also used the example `https://fastapi-users.github.io/fastapi-users/latest/configuration/oauth/#sqlalchemy_1` and there is no problem.

I would love to help you, but there is nothing I can do.

I am using `fastapi-users 13.0.0` `httpx-oauth 0.15.0`.

**Comment by leowenlu** [comment]
Thanks anyway, not sure where I got it wrong.
I am using the following. 
fastapi==0.112.1
fastapi-users==13.0.0
fastapi-users-db-sqlalchemy==6.0.1
httpx-oauth==0.15.1

I have tried this one, works good, but need write the whole user system instead:
https://github.com/tomasvotava/fastapi-sso/

**Comment by leowenlu** [comment, marked as answer]
Ok, after wasted a whole day, found out that as I switched to a new google account, I didn't switch on

Google People API on the new project.
